### PR TITLE
GIVCAMP-282 | React Player preview (light)

### PR DESCRIPTION
**Pull request by yvonnetangsu**
# READY FOR REVIEW

# Summary
- Add storyblok option to toggle embed media preview (light) mode

# Review By (Date)
- Retro

# Criticality
- 2

# Review Tasks

## Setup tasks and/or behavior to test

1. Go to https://deploy-preview-216--giving-campaign.netlify.app/stories/education-for-a-life-of-purpose
2. Look at videos in preview mode and compared to Kerri's mock (just look at the preview icon)
Screenshot 2024-01-29 at 1 57 04 PM
3. Inspect the preview div and see that there's an aria-label with something like "Play video......"
3. Click on preview and see video starts playing

# Associated Issues and/or People
- JIRA ticket(s) - GIVCAMP-282

### CODE REVIEW

File: components/EmbedMedia/EmbedMedia.tsx [hunk 1]
```diff
@@ -59,6 +71,13 @@ export const EmbedMedia = ({
               height="100%"
               url={mediaUrl}
               controls
+              playsinline
+              light={isPreview ? true : false}
+              playing={isPreview ? true : false}
+              playIcon={isPreview ? PlayPreviewIcon : undefined}
+              // This previewAriaLabel prop is not documented but it is in the React Player source code
+              previewAriaLabel={isPreview ? previewAriaLabel : undefined}
```
Comment (yvonnetangsu) [inline components/EmbedMedia/EmbedMedia.tsx:79]: SODA said this makes it ok accessible (it renders a clickable/focusable div). When I have time I'll open a PR to React Player to add a role="button" (SODA suggestion to make it fully accessible).

File: package.json [hunk 2]
```diff
@@ -40,7 +40,7 @@
     "framer-motion": "^10.16.4",
     "next": "13.4.9",
     "react-loading-skeleton": "^3.3.1",
-    "react-player": "^2.14.1",
+    "react-player": "^2.15.1",
```
Comment (yvonnetangsu) [inline package.json:43]: Update react player to latest which included this new previewAriaLabel prop

File: components/EmbedMedia/EmbedMedia.tsx (hunk 1 above)
Comment (sherakama) [reply components/EmbedMedia/EmbedMedia.tsx:79]: You can always just log a request or file a bug. It is nice of you to submit the PR but sometimes you can just let others know if you don't have the time.

**Comment by sherakama** [comment]
Screenshot 2024-04-05 at 09 39 10

This is looking and working just fine but I noticed that the double quotes are there. You have have to escape or encode the text going in.

**Review by sherakama — APPROVED** [review]
GTG

**Comment by yvonnetangsu** [comment]
> Screenshot 2024-04-05 at 09 39 10
> 
> This is looking and working just fine but I noticed that the double quotes are there. ~You may have to escape or encode the text going in.~
> 
> Looks like unicode FTW here.

Thanks for checking @sherakama ! 😄

File: components/FeaturedStories/FeatureMasonry.tsx [hunk 3]
```diff
@@ -92,8 +94,10 @@ export const FeatureMasonry = ({
         </div>
         <EmbedMedia
           mediaUrl={videoUrl}
+          isPreview
           className="lg:col-span-6 bg-black-70"
           aspectRatio="16x9"
+          previewAriaLabel={previewAriaLabel}
```
[the remark this answers is not in the document]
Comment (yvonnetangsu) [reply components/FeaturedStories/FeatureMasonry.tsx:100]: Yeah, it looks like they recently switched maintainer. I looked at the merged pull request, and looks like this prop was added fairly recently:
https://github.com/cookpete/react-player/pull/1705

I'll give them benefit of the doubt this time 😆

**Pull request merged**